main/settings/src/main/scala/sbt/Reference.scala

```diff
@@ -78,7 +78,7 @@ object Reference {
       case LocalRootProject    => "{<this>}<root>"
       case LocalProject(id)    => "{<this>}" + id
       case RootProject(uri)    => "{" + uri + " }<root>"
-      case ProjectRef(uri, id) => "{" + uri + "}" + id
+      case ProjectRef(uri, id) => s"""ProjectRef(uri("$uri"),"$id")"""
```

Comment:
The current shell notation covers `{<build-uri>}<project-id>/config:intask::key` and that it needs to be transcoded _without losing information_. The fact the the users never comes across non-current `ProjectRef` is simply untrue. Just start sbt on a fresh machine, you'll see it load your global plugins, and then your meta build. These need to be addressed.

The new function was added to display the `ProjectRef()` more concisely where possible. Simply removing the URL portion is not an option. You can make builds that switches together source dependencies with the same project ids.